3rdParty/OUIDataset/createOUIData.py

```diff
@@ -0,0 +1,113 @@
+import json
```

Comment:
Sure, why not

Reply:
Ok, I'll write the code and let you review

Reply:
@egecetin I refactored the python script and created a new one: [`create_oui_data.py`](https://github.com/egecetin/PcapPlusPlus/blob/e887c429f6747ea9745553f78cdec7e148f3747a/3rdParty/OUIDataset/create_oui_data.py). 
I also made a few updates to [your script](https://github.com/seladb/PcapPlusPlus/pull/904/commits/e887c429f6747ea9745553f78cdec7e148f3747a#diff-59a71286321350602915a1205009e69b0c768cfcf14b991b6ddd39d2337005ec):
- Fixed an encoding issue
- Removed the line which removes double quotes (`"`) - I don't think it's needed

If you download the latest `manuf.dat` and run both scripts the result will be similar except for one bug that I think I found in your script:

Some long mac addresses don't have a short address, for example:
```text
1C:87:74:00:00:00/28	PhilipsP	Philips Personal Health Solutions
1C:87:74:10:00:00/28	Sigfox
1C:87:74:20:00:00/28	Nichigak	Nichigaku
1C:87:74:30:00:00/28	SiloraR&	Silora R&D
1C:87:74:40:00:00/28	WeberMar	Weber Marking Systems GmbH
1C:87:74:50:00:00/28	Xiaoxing	Xiaoxinge (Tangshan) Electronic Technology Co., Ltd.
1C:87:74:60:00:00/28	Schawbel	Schawbel Technologies LLC
1C:87:74:70:00:00/28	IngBuero	Ing Buero Ziegler
1C:87:74:80:00:00/28	SurtecIn	Surtec Industries, Inc
1C:87:74:90:00:00/28	WideWorl	Wide World Trade HK ltd.
1C:87:74:A0:00:00/28	Nebbiolo	Nebbiolo Technologies
1C:87:74:B0:00:00/28	HABEYUSA	HABEY USA Inc.
1C:87:74:C0:00:00/28	NewNordi	New Nordic Engineering
1C:87:74:D0:00:00/28	Claber	Claber Spa
1C:87:74:E0:00:00/28	QuestInt	Quest Integrity
```

In your script those records are put under `ASUSTek COMPUTER INC.` but it has a different short mac:
```text
1C:87:2C	ASUSTekC	ASUSTek COMPUTER INC.
```

In my script they are put under a vendor with no name:
```json
    "1869684": {
        "vendor": "",
        "maskedFilters": [
            {
                "mask": 28,
                "vendors": {
                    "31368092319744": "Philips Personal Health Solutions",
                    "31368093368320": "Sigfox",
                    "31368094416896": "Nichigaku",
                    "31368095465472": "Silora R&D",
                    "31368096514048": "Weber Marking Systems GmbH",
                    "31368097562624": "Xiaoxinge (Tangshan) Electronic Technology Co., Ltd.",
                    "31368098611200": "Schawbel Technologies LLC",
                    "31368099659776": "Ing Buero Ziegler",
                    "31368100708352": "Surtec Industries, Inc",
                    "31368101756928": "Wide World Trade HK ltd.",
                    "31368102805504": "Nebbiolo Technologies",
                    "31368103854080": "HABEY USA Inc.",
                    "31368104902656": "New Nordic Engineering",
                    "31368105951232": "Claber Spa",
                    "31368106999808": "Quest Integrity"
                }
            }
        ]
    },
```

Where `1869684` corresponds to `1C:87:74`

Please review the script and let me know what you think

Reply:
Nice catch! A minor change might be adding "Unknown" to vendor field instead of empty for a case which short address matches but none of long addresses do not match. So `getVendorName` will return "Unknown" instead of empty string.

https://github.com/egecetin/PcapPlusPlus/blob/e745dc2d4c2be67265f0e0099c8ab890ce3e0bba/Common%2B%2B/src/OUILookup.cpp#L107